.github/workflows/build-and-test.yml

```diff
@@ -68,8 +70,30 @@ jobs:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-  notify-slack:
+  tag:
     needs: [ build, unit-tests-and-sonarqube, source-clear ]
+    uses: ./.github/workflows/tag.yml
+    if: success() && github.ref == 'refs/heads/main'
+    secrets: inherit
+
+  publish-library:
+    needs: [ tag ]
+    uses: ./.github/workflows/publish.yml
+    if: success() && github.ref == 'refs/heads/main'
+    secrets: inherit
+    with:
+      tag: ${{ needs.tag.outputs.tag }}
+
+  release-cli:
+    needs: [ tag ]
+    uses: ./.github/workflows/release-cli.yml
+    secrets: inherit
+    if: success() && github.ref == 'refs/heads/main'
+    with:
+      tag: ${{ needs.tag.outputs.tag }}
```

Comment:
Oh sorry, I probably missed that suggestion! 

I'd rather keep them as separate jobs. If nothing else, I like the diagram it produces 😀. I also have it so that the published library and publish cli jobs can run at the same time once the tag job completes (not that it saves that much time- they're both quick jobs). 
![image](https://github.com/DataBiosphere/java-pfb/assets/13254229/7af010b2-73e6-4935-b245-87b0bbcaddf2)